lib/maths/CTreeShapFeatureImportance.cc

```diff
@@ -112,23 +113,23 @@ void CTreeShapFeatureImportance::shapRecursive(const TTree& tree,
                                                const TDoubleVec& samplesPerNode,
                                                const CDataFrameCategoryEncoder& encoder,
                                                const CEncodedDataFrameRowRef& encodedRow,
-                                               SPath splitPath,
+                                               std::unique_ptr<SPath>& splitPath,
                                                std::size_t nodeIndex,
                                                double parentFractionZero,
                                                double parentFractionOne,
                                                int parentFeatureIndex,
                                                std::size_t offset,
                                                core::CDataFrame::TRowItr& row) const {
-
-    CTreeShapFeatureImportance::extendPath(splitPath, parentFractionZero,
+    std::unique_ptr<SPath> backupPath;
```

Comment:
My inclination would be to make this optional type, but pointer is ok. In any case, don't think stops us making splitPath a reference.

Reply:
Done